### PR TITLE
[PoC][API] Separate admin and shop docs

### PR DESCRIPTION
**Pull request by GSadee**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

Current problem:
- in both cases all schemas are displayed, only paths are limited based on the request url

`/new-api/admin/docs`:
Zrzut ekranu 2020-03-31 o 13 18 31

`/new-api/shop/docs`:
Zrzut ekranu 2020-03-31 o 13 18 45

### CODE REVIEW

File: src/Sylius/Bundle/ApiBundle/Swagger/DocumentationNormalizer.php [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+    {
+        return $this->decoratedNormalizer->supportsNormalization($data, $format);
+    }
+
+    public function normalize($object, string $format = null, array $context = [])
+    {
+        $docs = $this->decoratedNormalizer->normalize($object, $format, $context);
+
+        if ($this->isAdminDocs()) {
+            $docs['paths'] = array_filter($docs['paths']->getArrayCopy(), function ($path) {
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ApiBundle/Swagger/DocumentationNormalizer.php]:
```suggestion
            $docs['paths'] = array_filter($docs['paths']->getArrayCopy(), function (?string $path): bool {
```

File: src/Sylius/Bundle/ApiBundle/Swagger/DocumentationNormalizer.php [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+                            ],
+                        ],
+                    ],
+                ],
+            ];
+
+            return array_merge_recursive($tokenDocumentation, $docs);
+        }
+
+        $docs['paths'] = array_filter($docs['paths']->getArrayCopy(), function ($path) {
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ApiBundle/Swagger/DocumentationNormalizer.php]:
```suggestion
            $docs['paths'] = array_filter($docs['paths']->getArrayCopy(), function (?string $path): bool {
```

File: src/Sylius/Bundle/ApiBundle/Swagger/DocumentationNormalizer.php [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+        return $this->decoratedNormalizer->supportsNormalization($data, $format);
+    }
+
+    public function normalize($object, string $format = null, array $context = [])
+    {
+        $docs = $this->decoratedNormalizer->normalize($object, $format, $context);
+
+        if ($this->isAdminDocs()) {
+            $docs['paths'] = array_filter($docs['paths']->getArrayCopy(), function ($path) {
+                return substr($path, 0, 14) === '/new-api/admin';
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ApiBundle/Swagger/DocumentationNormalizer.php]: Probably should be parametrized

File: src/Sylius/Bundle/ApiBundle/Swagger/DocumentationNormalizer.php [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+    public function supportsNormalization($data, string $format = null): bool
+    {
+        return $this->decoratedNormalizer->supportsNormalization($data, $format);
+    }
+
+    public function normalize($object, string $format = null, array $context = [])
+    {
+        $docs = $this->decoratedNormalizer->normalize($object, $format, $context);
+
+        if ($this->isAdminDocs()) {
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ApiBundle/Swagger/DocumentationNormalizer.php]: What would you say, about creating two separate services, which will provide docs for admin and shop? Let's leave the content of this class as it is, just to extract the body of the if statement (env everything after it, but before `array_merge_recursive`. It should be easier to test and adjust the code if one would need it.

**Comment by pamil** [comment]
I guess this PR is no longer relevant after we've decided to remove prefixes in the new API.

**Pull request closed without merging**